packages/kbn-pm/src/commands/watch.ts

```diff
@@ -0,0 +1,110 @@
+import chalk from 'chalk';
+import { Observable, Subject } from 'rxjs';
+
+import { Project } from '../utils/project';
+import { topologicallyBatchProjects } from '../utils/projects';
+import { parallelizeBatches } from '../utils/parallelize';
+import { Command } from './';
+
+/**
+ * Name of the script in the package/project package.json file to run during `kbn watch`.
+ */
+const watchScriptName = 'kbn:watch';
+
+/**
+ * Number of milliseconds we wait before we fall back to the default watch handler.
+ */
+const defaultHandlerDelay = 3000;
+
+/**
+ * If default watch handler is used, then it's the number of milliseconds we wait for
+ * any build output before we consider watch task ready.
+ */
+const defaultHandlerReadinessTimeout = 2000;
+
+function getWatchHandlers(buildOutput$: Observable<string>) {
+  const typescriptHandler = buildOutput$
+    .first(data => data.includes('$ tsc'))
+    .map(() =>
+      buildOutput$
+        .first(data => data.includes('Compilation complete.'))
+        .mapTo('tsc')
+    );
+
+  const webpackHandler = buildOutput$
+    .first(data => data.includes('$ webpack'))
+    .map(() =>
+      buildOutput$.first(data => data.includes('Chunk Names')).mapTo('webpack')
+    );
+
+  const defaultHandler = Observable.of(void 0)
+    .delay(defaultHandlerReadinessTimeout)
+    .map(() =>
+      buildOutput$
```

Comment:
__note:__ this one will always fire: we can ignore it or introduce a local variable like `isHandled: boolean` that will be set by the first "handler" that default handler can check or do something even smarter with observables (but my Observables-fu is not there yet :)).

Reply:
This looks fine. It's easier to refactor when we've got some tests for it too :)